test/controllers/analytics.js

```diff
@@ -0,0 +1,65 @@
+const request = require('supertest');
+const chai = require('chai');
+
+const server = require('../../app');
+const helper = require('../helper');
+
+//Parent block
+describe('analytics API calls', function () {
```

Comment:
Since you're testing multiple routes, I would suggest using nested `describe` blocks to group your tests. This makes it easier to read and organize your tests, and to test a single route if needed.

```javascript
describe('Analytics API calls', function() {
  describe('POST /analytics/batchGet', function() {
    it('does something', function() { 
      ... 
    });
  ));

  // We only test this route.
  describe.only('POST /analytics/userActivity/:userId', function() {
    it('does something', function() { 
      ... 
    });
  ));
});
```

I see that the current tests don't use that structure though, so let's see what @martinbedouret thinks.

Reply:
agreed!